src/nvim/channel.c

```diff
@@ -517,6 +517,45 @@ uint64_t channel_from_stdio(bool rpc, CallbackReader on_output, const char **err
   return channel->id;
 }
 
+size_t newline_adjust(char **str)
+{
+#ifdef UNIX
+  return 0;
+#else
+  if (!str || !*str) {
+    return 0;
+  }
+  size_t c = 0;
+  char *p = *str;
+
+  while (*p++) {
+    c += *p == '\n' ? 1 : 0;
+  }
```

Comment:
Should I use `strcnt`?
https://github.com/neovim/neovim/blob/778541067a2f0ec3ef4bc1b07def646323e852a9/src/nvim/memory.c#L269-L286